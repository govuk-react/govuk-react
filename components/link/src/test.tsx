import * as React from 'react';
import { mount } from 'enzyme';

import { Link } from '.';

describe('Link', () => {
  it('basic view matches snapshot', () => {
    expect(
      mount(
        <Link muted={false} textColour={false} noVisitedState={false} href="#">
          Link example
        </Link>
      )
    ).toMatchSnapshot('plain link');
  });

  it('all options view matches snapshot', () => {
    expect(
      mount(
        <Link muted textColour noVisitedState href="#">
          Link example
        </Link>
      )
    ).toMatchSnapshot('link with all options');
  });
});
