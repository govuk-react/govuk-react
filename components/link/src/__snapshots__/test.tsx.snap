// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link all options view matches snapshot: link with all options 1`] = `
.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c0:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
  background-color: #ffdd00;
}

.c0:link {
  color: #1d70b8;
}

.c0:visited {
  color: #4c2c92;
}

.c0:hover {
  color: #2b8cc4;
}

.c0:active {
  color: #2b8cc4;
}

.c0:focus {
  color: #0b0c0c;
}

.c0:link,
.c0:visited,
.c0:hover,
.c0:active {
  color: #6f777b;
}

.c0:focus {
  color: #0b0c0c;
}

.c0:link,
.c0:visited,
.c0:hover,
.c0:active,
.c0:focus {
  color: #0b0c0c;
}

.c0:link {
  color: #1d70b8;
}

.c0:visited {
  color: #1d70b8;
}

.c0:hover {
  color: #2b8cc4;
}

.c0:active {
  color: #2b8cc4;
}

.c0:focus {
  color: #0b0c0c;
}

@media print {
  .c0 {
    font-family: sans-serif;
  }
}

@media print {
  .c0[href^="/"]::after,
  .c0[href^="http://"]::after,
  .c0[href^="https://"]::after {
    content: " (" attr(href) ")";
    font-size: 90%;
    word-wrap: break-word;
  }
}

@media print {
  .c0:link,
  .c0:visited,
  .c0:hover,
  .c0:active,
  .c0:focus {
    color: #000;
  }
}

<Link
  href="#"
  muted={true}
  noVisitedState={true}
  textColour={true}
>
  <a
    className="c0"
    href="#"
  >
    Link example
  </a>
</Link>
`;

exports[`Link basic view matches snapshot: plain link 1`] = `
.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c0:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
  background-color: #ffdd00;
}

.c0:link {
  color: #1d70b8;
}

.c0:visited {
  color: #4c2c92;
}

.c0:hover {
  color: #2b8cc4;
}

.c0:active {
  color: #2b8cc4;
}

.c0:focus {
  color: #0b0c0c;
}

@media print {
  .c0 {
    font-family: sans-serif;
  }
}

@media print {
  .c0[href^="/"]::after,
  .c0[href^="http://"]::after,
  .c0[href^="https://"]::after {
    content: " (" attr(href) ")";
    font-size: 90%;
    word-wrap: break-word;
  }
}

<Link
  href="#"
  muted={false}
  noVisitedState={false}
  textColour={false}
>
  <a
    className="c0"
    href="#"
  >
    Link example
  </a>
</Link>
`;
